File: python/n_ary_tree_postorder_traversal.py
# Definition for a Node.
from typing import List


class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children


class Solution:
    def __init__(self):
        self.result = []

    def postorderTraversal(self, root: Node) -> None:
        if root is None: return None
        for child in root.children:
            self.postorderTraversal(child)
        self.result.append(root.val)

    def postorder(self, root: Node) -> List[int]:
        self.postorderTraversal(root)
        return self.result

File: src/NAryTreePostorderTraversal.java
import java.util.ArrayList;
import java.util.List;

public class NAryTreePostorderTraversal {
    private static class Node {
        public int val;
        public List<Node> children;

        public Node() {}

        public Node(int _val) {
            val = _val;
        }

        public Node(int _val, List<Node> _children) {
            val = _val;
            children = _children;
        }
    }

    List<Integer> result = new ArrayList<>();

    public List<Integer> postorder(Node root) {
        postorderTraversal(root);
        return result;
    }

    private void postorderTraversal(Node root) {
        if (root == null) return;
        for (Node child : root.children) {
            postorderTraversal(child);
        }
        result.add(root.val);
    }
}
